presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/Multinode.java

```diff
@@ -0,0 +1,105 @@
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.time.Duration;
+
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.containers.BindMode.READ_ONLY;
+
+public final class Multinode
+        extends AbstractEnvironmentProvider
+{
+    private final PathResolver pathResolver;
+    private final DockerFiles dockerFiles;
+
+    private final String imagesVersion;
+    private final File serverPackage;
+
+    @Inject
+    public Multinode(
+            PathResolver pathResolver,
+            DockerFiles dockerFiles,
+            Standard standard,
+            Hadoop hadoop,
+            EnvironmentOptions environmentOptions)
+    {
+        super(ImmutableList.of(standard, hadoop));
+        this.pathResolver = requireNonNull(pathResolver, "pathResolver is null");
+        this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
+        imagesVersion = requireNonNull(environmentOptions.imagesVersion, "environmentOptions.imagesVersion is null");
+        serverPackage = requireNonNull(environmentOptions.serverPackage, "environmentOptions.serverPackage is null");
+    }
+
+    @Override
+    protected void extendEnvironment(Environment.Builder builder)
+    {
+        super.extendEnvironment(builder);
+
+        builder.configureContainer("presto-master", container -> {
+            container
+                    .withFileSystemBind(
+                            dockerFiles.getDockerFilesHostPath("conf/environment/multinode/multinode-master-jvm.config"),
+                            "/docker/presto-product-tests/conf/presto/etc/jvm.config",
+                            READ_ONLY)
+                    .withFileSystemBind(
+                            dockerFiles.getDockerFilesHostPath("conf/environment/multinode/multinode-master-config.properties"),
+                            "/docker/presto-product-tests/conf/presto/etc/config.properties",
+                            READ_ONLY);
+        });
+
+        builder.addContainer("presto-worker", createPrestoWorker());
+    }
+
+    @SuppressWarnings("resource")
+    private DockerContainer createPrestoWorker()
+    {
+        DockerContainer container = new DockerContainer("prestodev/centos6-oj8:" + imagesVersion)
+                .withFileSystemBind(dockerFiles.getDockerFilesHostPath(), "/docker/presto-product-tests", READ_ONLY)
+                .withFileSystemBind(pathResolver.resolvePlaceholders(serverPackage).toString(), "/docker/presto-server.tar.gz", READ_ONLY)
+                .withFileSystemBind(
+                        dockerFiles.getDockerFilesHostPath("conf/environment/multinode/multinode-worker-jvm.config"),
+                        "/docker/presto-product-tests/conf/presto/etc/jvm.config",
+                        READ_ONLY)
+                .withFileSystemBind(
+                        dockerFiles.getDockerFilesHostPath("conf/environment/multinode/multinode-worker-config.properties"),
```

Comment:
just worker-config?

Reply:
i sometimes open a file by its name only. That's why we have eg `HivePartitionManager` even though a `PartitionManager` class in `.hive` package could (in theory) be as unambiguous.